### PR TITLE
docs: add section to upgrade guide on using the unified location service

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The `LocationUpgradeModule` provides a unified way of handling URL updates across AngularJS and Angular. This adds a section to upgrade guide describing the router unification and usage.

### CODE REVIEW

File: aio/content/guide/upgrade.md [hunk 1]
```diff
@@ -842,7 +842,7 @@ After this, the service is injectable anywhere in AngularJS code:
 
 AngularJS and Angular both have separate routers for configuration, handling navigation, encoding and decoding URLs, redirects, and interacting with browser APIs. When migrating from AngularJS to Angular, you have the option of running both routers to handle navigating in different parts of your application. As your migrating from AngularJS, you want to take advantage of new APIs, and move as much of this functionality to Angular.
 
-To aid in your migration from AngularJS to Angular, a unified location service is provided to shift the routing responsibility previously handled by the `$location` provider in AngularJS to Angular. 
+To aid in your migration from AngularJS to Angular, a unified location service is provided to shift the routing responsibility previously handled by the [$location servce](https://docs.angularjs.org/api/ng/service/$location) in AngularJS to Angular.
```
Comment (gkalpak) [inline aio/content/guide/upgrade.md]: _servce_ --> _service_

File: aio/content/guide/upgrade.md [hunk 2]
```diff
@@ -838,6 +838,59 @@ After this, the service is injectable anywhere in AngularJS code:
 <code-example path="upgrade-module/src/app/a-to-ajs-providers/hero-detail.component.ts" header="hero-detail.component.ts">
 </code-example>
 
+## Using the Unified Angular Location Service
+
+AngularJS and Angular both have separate routers for configuration, handling navigation, encoding and decoding URLs, redirects, and interacting with browser APIs. When migrating from AngularJS to Angular, you have the option of running both routers to handle navigating in different parts of your application. As your migrating from AngularJS, you want to take advantage of new APIs, and move as much of this functionality to Angular.
```
Comment (gkalpak) [inline aio/content/guide/upgrade.md]: _your migrating_ --> _you are migrating_

File: aio/content/guide/upgrade.md [hunk 3]
```diff
@@ -838,6 +838,59 @@ After this, the service is injectable anywhere in AngularJS code:
...
+@NgModule({
+  imports: [
+    // Other NgModule imports...
+    LocationUpgradeModule.config()
+  ]
+})
+export class AppModule {}
+```
+
+The `LocationUpgradeModule.config()` method accepts a configuration object that allows you configure options including the `LocationStrategy` with the `useHash` property, and the URL prefix with `hashPrefix` property.
```
Comment (gkalpak) [inline aio/content/guide/upgrade.md]: - _allows you configure_ --> _allows you to configure_ (?)
- _with \`hashPrefix\` property_ --> _with the \`hashPrefix\` property_

**Review by gkalpak — CHANGES_REQUESTED** [review]
A couple of suggestions:

- Commit message typo:
  _docs(docs-infra): add common/upgrade/http to API package sources_ --> _[...] common/upgrade [...]_

- The commit that changes the parameter names in golden files should be squashed with the one the changes the actual parameters imo.

**Review by jasonaden — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: This includes a change to public API change in addition to docs that should not impact g3. Please run presubmit.

File: aio/content/guide/upgrade.md [hunk 4]
```diff
@@ -838,6 +838,59 @@ After this, the service is injectable anywhere in AngularJS code:
 <code-example path="upgrade-module/src/app/a-to-ajs-providers/hero-detail.component.ts" header="hero-detail.component.ts">
 </code-example>
 
+## Using the Unified Angular Location Service
+
+AngularJS and Angular both have separate routers for configuration, handling navigation, encoding and decoding URLs, redirects, and interacting with browser APIs. When migrating from AngularJS to Angular, you have the option of running both routers to handle navigating in different parts of your application. As you are migrating from AngularJS, you want to take advantage of new APIs, and move as much of this functionality to Angular.
+
```
Comment (jbogarthyde) [inline aio/content/guide/upgrade.md]: In AngularJS, the [$location service](https://docs.angularjs.org/api/ng/service/$location) handles all routing configuration and navigation, encoding and decoding of URLS, redirects, and interactions with browser APIs. Angular uses a different router for all of these tasks.

When you migrate from AngularJS to Angular, you have the option of running both routers to handle navigating in different parts of your application. *(I don't understand what it means to "run both routers" - does it mean you choose one for some tasks and the other for other tasks? I'm not clear if the unified loc service is itself the way you choose what to use where*)

To help with this, Angular provides the `LocationUpgradeModule`, a  *unified location service* that shifts routing responsibilities from the AngularJS `$location service` to the Angular router.

To take advantage of new APIs, you need to move as much functionality as possible to the Angular router. 

To use...

**Review by jbogarthyde — COMMENTED** [review]
Here's a bit of organizational change, based on my first reading.
It's not clear to me if "Unified Location Service" IS the "option to run both routers". Is it a tool for configuring which tasks should be handled by the old location service, and which should be handled by the new router? If so, let's state that up front.
We should talk, then I'd like to take another look at this.

File: aio/content/guide/upgrade.md [hunk 5]
```diff
@@ -838,6 +838,59 @@ After this, the service is injectable anywhere in AngularJS code:
 <code-example path="upgrade-module/src/app/a-to-ajs-providers/hero-detail.component.ts" header="hero-detail.component.ts">
 </code-example>
 
+## Using the Unified Angular Location Service
+
+In AngularJS, the [$location service](https://docs.angularjs.org/api/ng/service/$location) handles all routing configuration and navigation, encoding and decoding of URLS, redirects, and interactions with browser APIs. Angular uses its own underlying `Location` service for all of these tasks. 
+
+When you migrate from AngularJS to Angular you will want to move as much responsibility as possible to Angular, so that you can take advantage of new APIs. o help with the transition, Angular provides the `LocationUpgradeModule`. This module enables a _unified_ location service that shifts responsibilities from the AngularJS `$location` service to the Angular `Location` service.
+
```
Comment (jbogarthyde) [inline aio/content/guide/upgrade.md]: Typo:  To help with...

**Review by jbogarthyde — CHANGES_REQUESTED** [review]
Typo

File: aio/content/guide/upgrade.md (hunk 5 above)
Comment (brandonroberts) [reply aio/content/guide/upgrade.md]: Fixed

**Review by jbogarthyde — APPROVED** [review]
LGTM

**Pull request closed without merging**